### PR TITLE
Add sorting & filter functionality for both timesheets and team-members

### PR DESCRIPTION
**Pull request by tlorent**

### CODE REVIEW

File: components/team-members-overview/TeamMembersOverview.jsx [hunk 1]
```diff
@@ -126,6 +126,26 @@ class TeamMembersOverview extends React.Component {
               onAddTeamMember={this.addTeamMember}
             />)
           }
+          {teamMembers.map(({
+            firstName, lastName, id
+          }) => (
+            <React.Fragment key={id}>
```
Comment (boris-arkenaar) [inline components/team-members-overview/TeamMembersOverview.jsx]: No need for a fragment here

File: components/team-members-overview/TeamMembersOverview.jsx [hunk 2]
```diff
@@ -15,6 +16,11 @@ class TeamMembersOverview extends React.Component {
     this.props.requestTeamMembers();
   }
 
+  componentWillUnmount() {
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply components/team-members-overview/TeamMembersOverview.jsx]: Yeah, what's the idea here? You want to sort upon unmounting of the component?

File: components/team-members-overview/TeamMembersOverviewContainer.jsx [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+  teamMembers: teamMembersSelector(state),
+  sortBy: teamMembersSortBySelector(state)
 });
 
-const mapDispatchToProps = (dispatch) => bindActionCreators(actionCreators, dispatch);
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  requestTeamMembers,
+  requestTeamMembersSuccess,
+  postTeamMember,
+  postTeamMemberSuccess,
```
Comment (aronmartin) [inline components/team-members-overview/TeamMembersOverviewContainer.jsx]: Why are you binding all of these actions (success) to your component? 

If you are doing requests they should be handled in your saga. So if your request is successful then dispatch a success action in your saga

File: ducks/timesheets.js [hunk 4]
```diff
@@ -1,18 +1,58 @@
...
+  [timesheetsItemsSelector, timesheetsActiveFilterSelector],
+  (items, activeFilter) => items
+    .filter((item) => !activeFilter || item.employer === activeFilter)
+    .sort((a, b) => (a.startTime > b.startTime ? -1 : 1))
+);
+
+export const employersSelectors = createSelector(
+  timesheetsItemsSelector,
+  (items) => items.reduce((accumulator, item) => {
+    const exist = accumulator.some((employer) => employer.name === item.employer);
```
[the remark this answers is not in the document]
Comment (tlorent) [reply ducks/timesheets.js]: What about this (it's now for the names of the clients and in **clients.js**)?

```export const clientNamesSelector = createSelector(
  clientsItemsSelector,
  (items) => items.reduce((acc, item) => (
    acc.some((client) => client.name === item.clientName) ? [...acc] : [...acc, {
      id: item.id,
      name: item.clientName
    }]), [])
);```

File: ducks/team-members.js [hunk 5]
```diff
@@ -1,16 +1,56 @@
...
+  (teamMembers) => teamMembers.sortOrder
+);
+
+export const teamMembersSelector = createSelector(
+  [
+    teamMembersItemsSelector,
+    teamMembersSortBySelector,
+    teamMembersSortOrderSelector
+  ],
+  (items, sortBy, sortOrder) => [...items].sort((a, b) => {
```
[the remark this answers is not in the document]
Comment (tlorent) [reply ducks/team-members.js]: If we don't create a copy of ```items``` we are mutating the original by directly sorting it?

File: components/team-members-overview/TeamMembersOverviewContainer.jsx (hunk 3 above)
Comment (tlorent) [reply components/team-members-overview/TeamMembersOverviewContainer.jsx]: Of course, good call! I will edit this in other container components as well.

File: ducks/timesheets.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tlorent) [reply ducks/timesheets.js]: This method is now removed, not necessary anymore.

File: ducks/team-members.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (aronmartin) [reply ducks/team-members.js]: Don't think so. The only way you can mutate your data in your redux state is by returning a new state in your reducer.

[the remark this answers is not in the document]
Comment (tlorent) [reply ducks/team-members.js]: You're right, the state stays the same.

**Pull request merged**